staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -221,15 +224,15 @@ type ResourceEventHandler interface {
 // ResourceEventHandler.  This adapter does not remove the prohibition against
 // modifying the objects.
 type ResourceEventHandlerFuncs struct {
-	AddFunc    func(obj interface{})
+	AddFunc    func(obj interface{}, isInInitialList bool)
```

Comment:
This seems like it would require changes across the ecosystem. would it make sense to add another callback? That should be backwards compat for all cases except `ResourceEventHandlerFuncs{a,b,c}` which is typically a linter error anyways

Reply:
Yeah, the code can test for which version is passed in.

Reply:
That would make the tests less scary, for sure ;)

Reply:
yeah... it would be really great to not require a signature from all consumers

Reply:
OK I think this will accomplish that, the intermediate state was good to prove to me that that I got everything in this package passing this through correctly.

Still needs comment updates and tests.

And it needs to be used by one controller to prove / demonstrate how it works, what controller would you like me to update?